src/Elmish.WPF/BindingVmHelpers.fs

```diff
@@ -618,24 +622,25 @@ type Update
           this.Recursive(b.Get currentModel, b.Get newModel, b.Dispatch, b.Binding)
 
 
-type Get(nameChain: string) =
+type Get<'a>(nameChain: string) =
 
-  member _.Base (model: 'model, binding: BaseVmBinding<'model, 'msg>) =
+  member _.Base (model: 'model, binding: BaseVmBinding<'model, 'msg, 'a>): Result<'a, GetError> =
     match binding with
     | OneWay { OneWayData = d } -> d.Get model |> Ok
     | TwoWay b -> b.Get model |> Ok
     | OneWayToSource _ -> GetError.OneWayToSource |> Error
-    | OneWaySeq { Values = vals } -> vals.BoxedCollection() |> Ok
-    | Cmd cmd -> cmd |> box |> Ok
-    | SubModel { Vm = vm } -> vm.Value |> ValueOption.toObj |> box |> Ok
+    | OneWaySeq { Values = vals } -> vals.BoxedCollection() |> unbox |> Ok
+    | Cmd cmd -> cmd |> unbox |> Ok
+    | SubModel { Vm = vm } -> vm.Value |> ValueOption.map box |> ValueOption.toObj |> unbox |> Ok
```

Comment:
I find this line confusing because `'a` is used twice: once to say that `vm.Value` has type `'a voption` and once as the type of the value passed to `Ok`.

I think this works in practice because `'a` is only ever `obj`.

Can we make this clearer?  I think the fix is to have a second type parameter.

Similar situation for `SubModelWin` and (I think) for `SubModelSelectedItem`.

Reply:
> I think this works in practice because `'a` is only ever obj.

I'll write a test for different types. As far as I know, this process "normalizes" between nullable and non-nullable types (at least that's what it does with the type system.

Reply:
I added simple tests in #474 as that refactor is really where this becomes much easier to test.

Reply:
I extracted it into a pair of helper functions, `ValueOption.ofNull` and `ValueOption.toNull`. Basically they do the exact same thing as `ValueOption.ofObj` and `ValueOption.toObj` except they also handle non-nullable types.